### PR TITLE
CXSPA-4049 Add tests for payment verification process

### PR DESCRIPTION
**Pull request by rmch91**
Closes https://jira.tools.sap/browse/CXSPA-4049

### CODE REVIEW

File: integration-libs/opf/base/root/services/opf-state-persistence.service.spec.ts [hunk 1]
```diff
@@ -1 +1,119 @@
...
+    );
+
+    service['getOpfState']().subscribe((state) => {
+      expect(state).toEqual({ metadata: mockOpfMetadata });
+      done();
+    });
+  });
+
+  it('should update OpfPaymentMetadataStoreService when onRead is called', () => {
+    const mockOpfMetadata: OpfPaymentMetadata = {
```
Comment (Matejk00) [inline integration-libs/opf/base/root/services/opf-state-persistence.service.spec.ts]: Can we declare this const more at the top of the test scenarios? So it's no need to re-declare this property each time.

File: integration-libs/opf/base/root/services/opf-payment-metadata-store.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    service.getOpfMetadataState().subscribe((state) => {
+      expect(state).toEqual(state);
+      done();
+    });
+  });
+
+  it('should update opfPaymentMetadataState with the given payload', () => {
+    const state: OpfPaymentMetadata = {
+      isPaymentInProgress: false,
```
Comment (Matejk00) [inline integration-libs/opf/base/root/services/opf-payment-metadata-store.service.spec.ts:54]: The same case here.

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: integration-libs/opf/base/root/services/opf-payment-metadata-store.service.spec.ts (hunk 2 above)
Comment (rmch91) [reply integration-libs/opf/base/root/services/opf-payment-metadata-store.service.spec.ts:54]: done

File: integration-libs/opf/base/root/services/opf-state-persistence.service.spec.ts (hunk 1 above)
Comment (rmch91) [reply integration-libs/opf/base/root/services/opf-state-persistence.service.spec.ts]: Changed

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**